code/script.js

```diff
@@ -244,100 +296,155 @@ const generateBoard = () => {
+        alert(`✔ Yes, the person has ${value} hair! Good guess.
+🤖 Keeping every ${value} haired ones.`)
+      } else {
+        alert(`❌ Nope, the person doesn't have ${value} hair! Guess again.
+🤖 Removing those with ${value} hair.`)
+        }
+    } else if (category === 'eye color') {
+      if (guess) {
+        alert(`✔ Yes, gaze into their ${value} eyes! Good guess.
+🤖 Keeping the ${value} eyed ones.`)
+      } else {
+        alert(`❌ Nope, the person doesn't have ${value} eyes!
+🤖 Removing everyone with ${value} eyes.`)
+        }
     }
+  if (guess) {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
   }
-
-  // filter to keep or remove based on the keep variable.
-  /* charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-    or 
-    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value) */
-
-  // Invoke a function to redraw the board with the remaining people.
+  else {
+    charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  }
+  generateBoard()
 }
 
-// when clicking guess, the player first have to confirm that they want to make a guess.
+// Confirm with the user if they meant to guess
 const guess = (suspect) => {
-  // store the interaction from the player in a variable.
-  // remember the confirm() ?
-  // If the player wants to guess, invoke the checkMyGuess function.
+  confirm(`🙈 You're about to guess on ${suspect}. Are you sure?`)
+  if (confirm) {
+    checkMyGuess(suspect)
+  }
 }
 
-// If you confirm, this function is invoked
+// If the user confirms their guess,
+// invoke a prompt,
+// and depending on result change the page
 const checkMyGuess = (suspect) => {
-  // 1. Check if the suspect is the same as the secret person's name
-  // 2. Set a Message to show in the win or lose section accordingly
-  // 3. Show the win or lose section
-  // 4. Hide the game board
+  if (suspect === secret.name) {
+    winOrLoseText.innerHTML = `Yep, it was <em>${suspect}</em>! Congratulations, detective! 🎉`
+  }
+  else {
+    winOrLoseText.innerHTML = `Close, yet so far away. It was <em>${secret.name}</em>, not <em>${suspect}</em>. Better luck next time. 🎭`
+  }
+  winOrLose.style.display = 'flex'
+  board.style.display = 'none'
 }
 
-// Invokes the start function when website is loaded
+// Start the game on load
 start()
 
-// All the event listeners
+// If the user clicks the Play Again button,
+// invoke the start function
+playAgain.addEventListener('click', start)
+
+// If the user clicks the Restart button,
+// invoke the start function
 restartButton.addEventListener('click', start)
+
+// If the user clicks the Find Out button,
+// invoke the checkQuestion function
+findOut.addEventListener('click', checkQuestion)
+
+// If the user selects a value from the drop-down,
+// pass its value to the selectQuestion function
+questions.addEventListener('change', () => selectQuestion())
```

Comment:
I realize this is how we wrote it in the Q/A as well, but since we no longer pass along an argument, we can write this like the others:
`questions.addEventListener('change', selectQuestion)`